tests/testthat/test-lancelot-ifr.R

```diff
@@ -240,3 +240,45 @@ test_that("Validate IFR excluding immunity inputs", {
   expect_error(lancelot_ifr_excl_immunity(1, list(p)),
                "Cannot currently compute IFR if any of 'hosp_transmission'")
 })
+
+
+test_that("Severity by age is calculated parametrically", {
+
+  helper <- function(p, i) {
+    mod <- lancelot$new(p, 0, 1, seed = 1L)
+    info <- mod$info()
+    state <- lancelot_initial(info, 1, p)
+
+    index_i <- array(info$index[[i]], info$dim[[i]])
+
+    mod$update_state(state = state)
+    mod$set_index(info$index[[i]])
+
+    y <- mod$simulate(seq(0, 400, by = 4))
+
+    expect_equal(length(y), prod(info$dim[[i]]) * 101)
+
+    y <- array(y, c(info$dim[[i]], 101))
+  }
+
+  p <- lancelot_parameters(1, "uk", carehome_beds = 0)
+
+  ihr <- p$p_C_step * p$p_H_step * (1 - p$p_G_D_step)
+  hfr <- (1 - p$p_ICU_step) * p$p_H_D_step +
+    p$p_ICU_step * (p$p_ICU_D_step + (1 - p$p_ICU_D_step) * p$p_W_D_step)
+  ifr <- ihr * hfr + p$p_C_step * p$p_H_step * p$p_G_D_step
+
+  # Severity is expected to be highest for the 80+
+  y <- helper(p, "ifr_age")[c(1:17), 101]
+  x <- ifr[which(!is.na(y))]
+  expect_true(sum(y, na.rm = TRUE) == sum(x))
+
+  y <- helper(p, "ihr_age")[c(1:17), 101]
+  x <- ihr[which(!is.na(y))]
+  expect_true(sum(y, na.rm = TRUE) == sum(x))
+
+  y <- helper(p, "hfr_age")[c(1:17), 101]
+  x <- hfr[which(!is.na(y))]
+  expect_true(sum(y, na.rm = TRUE) == sum(x))
```

Comment:
I don't really follow the logic of these tests. Why are we summing?

Reply:
Also these tests do not use the functions in lancelot-ifr.R, so belong somewhere else (test-lancelot-check fits better)